### PR TITLE
IDMP-433: Add idmp-sub:hasMoiety

### PR DESCRIPTION
**Pull request by aliariff**

### CODE REVIEW

**Review by Heiner-Oberkampf — DISMISSED** [review]
looks good

File: etc/transformation/GSRS/gsrs-public-data-substances.rqg [hunk 1]
```diff
@@ -21,17 +21,26 @@ GENERATE {
...
         ?hasMolecularFormulaByMoietyProperty ?structureFormula ;
-        idmp-sub:hasStereochemistry ?Stereochemistry ;
-        idmp-sub:hasMolecularWeight ?structureMolecularWeight ;
+        idmp-sub:hasDefiningStereochemistry ?Stereochemistry ;
+        idmp-sub:hasDefiningMolecularWeight ?structureMolecularWeight ;
+        idmp-sub:hasMoiety ?MolecularStructureMoiety ;
+        .
+
+    # Molecular Structure Moiety
+    ?MolecularStructureMoiety a idmp-sub:MolecularStructure ;
```
Comment (mereolog) [inline etc/transformation/GSRS/gsrs-public-data-substances.rqg:39]: This is minor, by why a variable instantiating idmp-sub:MolecularStructure is called ?MolecularStructureMoiety?

File: etc/transformation/GSRS/gsrs-public-data-substances.rqg [hunk 2]
```diff
@@ -21,17 +21,26 @@ GENERATE {
...
     ?MolecularStructure a idmp-sub:MolecularStructure ;
         idmp-sub:hasSimplifiedMolecularInputLineEntrySpecification ?structureSmilesString ;
-        idmp-sub:hasMolecularFormula ?structureFormula ;
+        idmp-sub:hasDefiningMolecularFormula ?structureFormula ;
         ?hasMolecularFormulaByMoietyProperty ?structureFormula ;
-        idmp-sub:hasStereochemistry ?Stereochemistry ;
-        idmp-sub:hasMolecularWeight ?structureMolecularWeight ;
+        idmp-sub:hasDefiningStereochemistry ?Stereochemistry ;
+        idmp-sub:hasDefiningMolecularWeight ?structureMolecularWeight ;
+        idmp-sub:hasMoiety ?MolecularStructureMoiety ;
```
Comment (mereolog) [inline etc/transformation/GSRS/gsrs-public-data-substances.rqg:35]: This looks wrong.
In the IDMP ontology all these properties, except for idmp-sub:hasDefiningStereochemistry are used for substances, and not molecular structures. So IDMP ontology has it, for example, that idmp-sub:SingleSubstance is a subclass of the "'has defining molecular formula' only 'molecular formula'" restriction.
None of these properties are used for idmp-sub:MolecularStructure.

File: etc/transformation/GSRS/gsrs-public-data-substances.rqg [hunk 3]
```diff
@@ -21,17 +21,26 @@ GENERATE {
...
+        .
+
+    # Molecular Structure Moiety
+    ?MolecularStructureMoiety a idmp-sub:MolecularStructure ;
+        idmp-sub:hasSimplifiedMolecularInputLineEntrySpecification ?moietySmilesString ;
+        idmp-sub:hasDefiningMolecularFormula ?moietyFormula ;
+        idmp-sub:hasDefiningStereochemistry ?moietyStereochemistryIDMP ;
+        idmp-sub:hasDefiningMolecularWeight ?moietyMolecularWeight ;
```
Comment (mereolog) [inline etc/transformation/GSRS/gsrs-public-data-substances.rqg:43]: See my other comment above.

File: etc/transformation/GSRS/gsrs-public-data-substances.rqg (hunk 2 above)
Comment (Heiner-Oberkampf) [reply etc/transformation/GSRS/gsrs-public-data-substances.rqg:35]: @aliariff we discussed in the weekly today with Thomas and yes, these properties should be updated as indicated by Pawel. E.g., put "[hasDefiningMolecularWeight](https://spec.pistoiaalliance.org/idmp/ontology/ISO/ISO11238-Substances/hasDefiningMolecularWeight)" to the substance object and remove them from the "molecular structure" objects.

**Review by Heiner-Oberkampf — APPROVED** [review]

**Review by mereolog — APPROVED** [review]

**Pull request merged**